File: ai/repositories/profileRepository.js
import usersModel from '../../models/usersModel.js';

export class ProfileRepository {
  async findUserById(userId) {
    return usersModel.findUserById(userId);
  }

  async findUserByEmail(email) {
    return usersModel.findUserByEmail(String(email || '').toLowerCase());
  }
}

File: ai/repositories/profile_repository.py
from bson import ObjectId
from config.db import get_db


class ProfileRepository:
    def _col(self):
        return get_db()["users"]

    def find_user_by_id(self, user_id):
        try:
            oid = ObjectId(str(user_id))
        except Exception:
            return None
        doc = self._col().find_one({"_id": oid})
        if doc:
            doc.pop("password", None)
        return doc

    def find_user_by_email(self, email: str):
        doc = self._col().find_one({"email": str(email or "").lower().strip()})
        if doc:
            doc.pop("password", None)
        return doc
